src/report.rs

```diff
@@ -138,6 +166,26 @@ impl ContextReportExt for Context {
         let data_container = self.get_data_container_mut(ReportPlugin);
         &mut data_container.config
     }
+
+    /// Adds a periodic report to the context. The report will be output every `report_period` time
+    /// and will print out the output from the `report_output_fcn` function.
+    fn add_periodic_report<T: Report + 'static, C: Fn(&Context) -> T + 'static>(
```

Comment:
I think this would actually be better as a more generic `add_periodic_plan_with_priority`: would you mind moving it there instead? you could still have a test/example for adding the report and then scheduling it. 

Also, I think then C can just be generic then (like other plans).

LGTM with that change